caikit/core/model_management/local_model_trainer.py

```diff
@@ -78,25 +78,18 @@ def __init__(
         ):
             super().__init__(
                 trainer_name=trainer_name,
-                training_id=str(uuid.uuid4()),
+                training_id=external_training_id or str(uuid.uuid4()),
                 save_with_id=save_with_id,
                 save_path=save_path,
                 model_name=model_name,
+                use_reversible_hash=external_training_id is None,
             )
-            # 🌶️🌶️🌶️ For the external training id override, we don't want to include the
-            # reversible hash bit on the id. Therefore, we have to re-do the id and save
-            # path stuff done in the super() init here instead
-            if external_training_id is not None:
-                # No extra hash bit
-                self._id = external_training_id
-                self._save_path = self._save_path_with_id(
-                    save_path, save_with_id, external_training_id, model_name
-                )
-
             self._module_class = module_class
 
             # Placeholder for the time when the future completed
             self._completion_time = None
+            # Set the submission time as right now. (Maybe this should be supplied instead?)
+            self._submission_time = datetime.now()
```

Comment:
Maybe this is not applicable for local model trainer, but I'm curious how the submission time will work once we have multiple training jobs queued up that may take a long time, would they all have same submission time even though the training potentially won't start until later?

Reply:
Right, so for other remote training systems that do their own queuing the trainer implementation would presumably ask that system what time the training was submitted at, and it'd be up to that system to provide a correct value